okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java

```diff
@@ -954,19 +959,38 @@ private void proxyInternal(Iterator<ModuleInstance> it,
     );
   }
 
+  static Buffer compressInternalResponse(RoutingContext ctx, Buffer respBuf) {
+    String acceptEncoding = ctx.request().getHeader("Accept-Encoding");
+    int sz = respBuf.length();
+    if (acceptEncoding != null && acceptEncoding.contains("gzip")
+        && sz > CONTENT_COMPRESS_MIN_SIZE) {
+      try {
+        ByteArrayOutputStream dst = new ByteArrayOutputStream();
```

Comment:
Seems like `dst` should be some other name like `outputStream` or similar.

Reply:
I dunno.. both `src` and `dst` are outputstreams. `uncompressedOutputStream` and `compressedOutputStream`??